segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/delegationtoken/TokenVerifierImpl.java

```diff
@@ -75,6 +75,14 @@ public JsonWebToken verifyToken(@NonNull String resource, String token, @NonNull
      * @return
      */
     private boolean resourceMatchesClaimKey(String claimKey, String resource) {
+        log.trace("claimKey = {}, resourceKey = {}", claimKey, resource);
+        if (claimKey.startsWith("prn::")) {
+            claimKey = claimKey.replace("prn::/", "")
+                    .replace("prn::", "")
```

Comment:
This is just to confirm my understanding:   the fact that the delegation token deals with resource strings that are in the same format as what goes on in the controller side authorization is just a coincidence, correct?

Meaning, in theory what goes on in the delegation token _could_ have been completely different (though why re-invent the wheel basically).  is that correct?

Reply:
To provide a bit more context: For Segment store operations like reading from stream and appending to a stream, the user must have corresponding read/write access. The delegation token generated by the Controller has a "claim" of the form `<resource string>=<READ | READ_WRITE>`  that tells the segment store that the bearer of the token is allowed to perform the specified action. 

To answer your question, yes the delegation token could use a completed different format for the authorization key, but why re-invent the wheel.

Reply:
Got it. Thanks for confirming.